client.go

```diff
@@ -494,22 +495,27 @@ func (c *UncachedHTTPClient) getClient() (HTTP, error) {
 }
 
 func (c *UncachedHTTPClient) getHTTPData(method string, path string, body url.Values, data interface{}) error {
-	r, err := http.NewRequest(method, "https://"+c.Host+path, bytes.NewBufferString(body.Encode()))
-	if err != nil {
-		return err
-	}
-
 	if len(c.AuthHandlers) == 0 {
 		return errNoAuth
 	}
 
 	authRequestAttempted := false
+	attemptedAuthTypes := []string{}
+
 	for _, authHandler := range c.AuthHandlers {
-		authToken, clientOverride := authHandler()
+		authToken, authType, clientOverride := authHandler()
 		if authToken == "" {
 			continue
 		}
 		authRequestAttempted = true
+		attemptedAuthTypes = append(attemptedAuthTypes, authType)
+
+		// Create the request per authHandler to prevent body from being reused between requests.
+		// This is due to the body being non-reusable after the first read.
+		r, err := http.NewRequest(method, "https://"+c.Host+path, bytes.NewBufferString(body.Encode()))
```

Comment:
Q: Will `path` always have `/` as a prefix, or does `c.Host` always have `/` as a suffix?

Reply:
taking a look at some of the handlers, it seems like the path always contains the `/` as a prefix: ex. https://github.com/pinterest/knox/blob/master/client.go#L366